File: app/src/main/java/com/fishweeb/practical/DamageStatButton.java
package com.fishweeb.practical;

import android.graphics.Bitmap;
import android.graphics.Color;

public class DamageStatButton extends StatButtonEntity
{
    public void OnClickFunction()
    {
        if (PlayerInfo.Instance.GetMoney() >= Cost)
        {
            PlayerInfo.Instance.GetMainWeapon().SetWeaponDamage(PlayerInfo.Instance.GetMainWeapon().GetWeaponDamage() + AddValue);
            Value += AddValue;
            PlayerInfo.Instance.MinusMoney(Cost);
            GameSystem.Instance.SaveEditBegin();
            GameSystem.Instance.SetIntInSave("maingunshootdamage",PlayerInfo.Instance.GetMainWeapon().GetWeaponDamage());
            GameSystem.Instance.SetIntInSave("money",PlayerInfo.Instance.GetMoney());
            GameSystem.Instance.SaveEditEnd();
        }
    }

    public void OffClickFunction()
    {

    }

    public static DamageStatButton Create(float xPos, float yPos, int value, int addvalue, int cost, float buttonSize, float textSize)
    {
        DamageStatButton result = new DamageStatButton();
        result.Pos.x = xPos;
        result.Pos.y = yPos;
        result.bmp = Bitmap.createScaledBitmap(ResourceManager.Instance.GetBitmap(R.drawable.plusbutton),(int)buttonSize,(int)buttonSize,true);
        result.coinBitMap = Bitmap.createScaledBitmap(ResourceManager.Instance.GetBitmap(R.drawable.coin),(int)buttonSize,(int)buttonSize,true);
        result.Value =value;
        result.AddValue = addvalue;
        result.Cost = cost;
        result.paint.setColor(Color.BLACK);
        result.paint.setTextSize(textSize);
        EntityManager.Instance.AddEntity(result);
        return result;
    }
}

File: app/src/main/kotlin/com/fishweeb/practical/DamageStatButton.kt
package com.fishweeb.practical

import android.graphics.Bitmap
import android.graphics.Color

class DamageStatButton : StatButtonEntity() {
    override fun OnClickFunction() {
        if (PlayerInfo.Companion.Instance.GetMoney() >= Cost) {
            PlayerInfo.Companion.Instance.GetMainWeapon()!!.SetWeaponDamage(
                PlayerInfo.Companion.Instance.GetMainWeapon()!!.GetWeaponDamage() + AddValue
            )
            Value += AddValue
            PlayerInfo.Companion.Instance.MinusMoney(Cost)
            GameSystem.Companion.Instance.SaveEditBegin()
            GameSystem.Companion.Instance.SetIntInSave(
                "maingunshootdamage", PlayerInfo.Companion.Instance.GetMainWeapon()!!
                    .GetWeaponDamage()
            )
            GameSystem.Companion.Instance.SetIntInSave(
                "money",
                PlayerInfo.Companion.Instance.GetMoney()
            )
            GameSystem.Companion.Instance.SaveEditEnd()
        }
    }

    override fun OffClickFunction() {
    }

    companion object {
        fun Create(
            xPos: Float,
            yPos: Float,
            value: Int,
            addvalue: Int,
            cost: Int,
            buttonSize: Float,
            textSize: Float
        ): DamageStatButton {
            val result = DamageStatButton()
            result.Pos.x = xPos
            result.Pos.y = yPos
            result.bmp = Bitmap.createScaledBitmap(
                ResourceManager.Companion.Instance.GetBitmap(R.drawable.plusbutton)!!,
                buttonSize.toInt(),
                buttonSize.toInt(),
                true
            )
            result.coinBitMap = Bitmap.createScaledBitmap(
                ResourceManager.Companion.Instance.GetBitmap(R.drawable.coin)!!,
                buttonSize.toInt(),
                buttonSize.toInt(),
                true
            )
            result.Value = value
            result.AddValue = addvalue
            result.Cost = cost
            result.paint.color = Color.BLACK
            result.paint.textSize = textSize
            EntityManager.Companion.Instance.AddEntity(result)
            return result
        }
    }
}